resources/maskinporten_clients.py

```diff
@@ -140,15 +141,92 @@ def list_clients(env: MaskinportenEnvironment, auth_info: AuthInfo = Depends()):
     return clients
 
 
+@router.delete(
+    "/{env}/{client_id}",
+    status_code=status.HTTP_200_OK,
+    response_model=DeleteMaskinportenClientOut,
+    responses=error_message_models(
+        status.HTTP_400_BAD_REQUEST,
+        status.HTTP_401_UNAUTHORIZED,
+        status.HTTP_403_FORBIDDEN,
+        status.HTTP_404_NOT_FOUND,
+        status.HTTP_422_UNPROCESSABLE_ENTITY,
+    ),
+)
+def delete_client(
+        env: MaskinportenEnvironment,
+        client_id: str = Path(..., regex=r"^[0-9a-f-]+$"),
+        auth_info: AuthInfo = Depends(),
+):
+    authorize(
+        auth_info,
+        scope="maskinporten:client:write",
+        resource=f"maskinporten:client:{env}-{client_id}",
+    )
+
+    try:
+        maskinporten_client = MaskinportenClient(env)
+    except UnsupportedEnvironmentError as e:
+        raise ErrorResponse(status.HTTP_400_BAD_REQUEST, str(e))
+
+    try:
+        maskinporten_client.get_client(client_id)
+    except requests.HTTPError as e:
+        if e.response.status_code == status.HTTP_404_NOT_FOUND:
+            raise ErrorResponse(
+                status.HTTP_404_NOT_FOUND, f"No client with ID {client_id}"
+            )
+        raise
+
+    try:
+        # Search for active keys associated with client
+        existing_jwks = self.get_client_keys(client_id).json().get("keys", [])
```

Comment:
`get_client_keys` er må kalles gjennom `maskinporten_client` i stedet for `self`.